src/rdkafka_request.c

```diff
@@ -1211,6 +1256,20 @@ void rd_kafka_JoinGroupRequest (rd_kafka_broker_t *rkb,
                            rk->rk_conf.max_poll_interval_ms,
                            rk->rk_conf.group_session_timeout_ms);
 
+
+        if (ApiVersion < 5 &&
+            rk->rk_conf.group_instance_id &&
+            rd_interval(&rkb->rkb_suppress.unsupported_kip345,
+                        /* at most once per day */
+                        (rd_ts_t)86400 * 1000 * 1000, 0) > 0)
+                rd_rkb_log(rkb, LOG_NOTICE, "MAXPOLL",
+                           "Broker does not support KIP-345 "
```

Comment:
Isn't this an error? The user configured the consumer to be a static member but that can't be realized due to outdated broker version.
I'd say you want to raise a consumer error and back off the next join 30s or so.

Reply:
I'd say max.poll is not so much of a feature as it is a control mechanism for the client to monitor the application, while static membership is a feature an application might want to rely on.

But we can leave it as-is for now and improve in the future if necessary.